### PR TITLE
refactor(rln): remove kilic lib

### PR DESCRIPTION
**Pull request by s1fr0**
This PR addresses https://github.com/status-im/nwaku/issues/1314 by removing support to Kilic's lib.

I removed from the Makefile a seemingly orphan `clean-common` target called by `clean`.

### CODE REVIEW

File: tests/all_tests_v2.nim [hunk 1]
```diff
@@ -1,58 +1,9 @@
...
-  ./v2/test_waku_keepalive,
-  ./v2/test_namespacing_utils,
-  ./v2/test_waku_dnsdisc,
-  ./v2/test_waku_discv5,
-  ./v2/test_enr_utils,
-  ./v2/test_peer_exchange,
-  ./v2/test_waku_noise,
-  ./v2/test_waku_noise_sessions,
-  # Utils
-  ./v2/test_utils_keyfile
```
Comment (rymnc) [inline tests/all_tests_v2.nim:53]: These should be included, I guess

File: tests/v2/test_waku_rln_relay.nim [hunk 2]
```diff
@@ -512,71 +460,39 @@ suite "Waku rln relay":
...
+                          nullifier: nullifier,
+                          rlnIdentifier: rlnIdentifier)
+      protobuf = rateLimitProof.encode()
+      decodednsp = RateLimitProof.init(protobuf.buffer)
 
-      check:
-        decodednsp.isErr == false
-        decodednsp.value == rateLimitProof
+    check:
+      decodednsp.isErr == false
```
Comment (rymnc) [inline tests/v2/test_waku_rln_relay.nim:494]:
```suggestion
      decodednsp.isErr() == false
```

File: waku/v2/protocol/waku_rln_relay/waku_rln_relay_constants.nim [hunk 3]
```diff
@@ -22,12 +22,10 @@ const
...
-
-when defined(rlnzerokit):
-  const
-    # The relative folder where the circuit, proving and verification key for RLN can be found
-    # Note that resources has to be compiled with respect to the above MerkleTreeDepth
-    RlnResourceFolder* = "vendor/zerokit/rln/resources/tree_height_" & $MerkleTreeDepth & "/"
+const
+  # The relative folder where the circuit, proving and verification key for RLN can be found
+  # Note that resources has to be compiled with respect to the above MerkleTreeDepth
+  RlnResourceFolder* = "vendor/zerokit/rln/resources/tree_height_" & $MerkleTreeDepth & "/"
```
Comment (rymnc) [inline waku/v2/protocol/waku_rln_relay/waku_rln_relay_constants.nim:28]:
```suggestion
when defined(rln):
  const
    # The relative folder where the circuit, proving and verification key for RLN can be found
    # Note that resources has to be compiled with respect to the above MerkleTreeDepth
    RlnResourceFolder* = "vendor/zerokit/rln/resources/tree_height_" & $MerkleTreeDepth & "/"
```

File: waku/v2/protocol/waku_rln_relay/rln.nim [hunk 4]
```diff
@@ -9,10 +9,7 @@ import
   os,
   waku_rln_relay_types
 
-when defined(rln) or (not defined(rln) and not defined(rlnzerokit)):
-  const libPath = "vendor/rln/target/debug/"
-when defined(rlnzerokit):
-  const libPath = "vendor/zerokit/target/release/"
+const libPath = "vendor/zerokit/target/release/"
```
Comment (rymnc) [inline waku/v2/protocol/waku_rln_relay/rln.nim:12]:
```suggestion
when defined(rln):
  const libPath = "vendor/zerokit/target/release/"
```

File: waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim [hunk 5]
```diff
@@ -48,132 +68,54 @@ proc toBuffer*(x: openArray[byte]): Buffer =
...
-      secret: array[32, byte]
-      public: array[32, byte]
-    for (i, x) in secret.mpairs: x = generatedKeys[i]
-    for (i, x) in public.mpairs: x = generatedKeys[i+32]
-
-    var
-      keypair = MembershipKeyPair(idKey: secret, idCommitment: public)
-
-    return ok(keypair)
+proc createRLNInstanceLocal(d: int = MerkleTreeDepth): RLNResult =
```
Comment (rymnc) [inline waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim:71]: This should defined only if `rln` is defined, right?

**Review by rymnc — APPROVED** [review]
LGTM, minor nits! Thanks :)

File: waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim [hunk 6]
```diff
@@ -26,10 +26,30 @@ import
...
+    rlnRelayMembershipIndex*: uint
+    rlnRelayEthContractAddress*: string
+    rlnRelayEthClientAddress*: string
+    rlnRelayEthAccountPrivateKey*: string
+    rlnRelayEthAccountAddress*: string
+    rlnRelayCredPath*: string
+    rlnRelayCredentialsPassword*: string
+    
+type 
+  MessageValidationResult* {.pure.} = enum
```
Comment (staheri14) [inline waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim:42]: I see this one is moved from the waku_rln_relay_types module; curious to know why, any style guide?

**Review by staheri14 — APPROVED** [review]
LGTM!

**Comment by LNSD** [comment]
> I removed from the Makefile a seemingly orphan `clean-common` target called by `clean`.

I don't see the `Makefile` changes in the PR 👀

**Review by LNSD — APPROVED** [review]
LGTM ✅

> "Deleted code is debugged code."  - Jeff Sickel

**Pull request merged**